src/webui/webapplication.cpp

```diff
@@ -224,12 +224,15 @@ void WebApplication::action_public_images()
 void WebApplication::action_query_torrents()
 {
     CHECK_URI(0);
-    const QStringMap& gets = request().gets;
-
-    print(btjson::getTorrents(
-        gets["filter"], gets["category"], gets["sort"], gets["reverse"] == "true",
-        gets["limit"].toInt(), gets["offset"].toInt()
-        ), Http::CONTENT_TYPE_JSON);
+    const RawStringMap& gets = request().gets;
+
+    print(btjson::getTorrents(gets["filter"],
+                              QString::fromUtf8(gets["category"]),
```

Comment:
The PR changes the way GET query parameters are decoded. RequestParser used to convert them directly into QString expecting the percent-encoded byte sequence to be UTF-8. On the other hand, tracker announce url can contain arbitrary percent-encoded 20-byte sequences in info_hash and peer_id parameters that have no valid UTF-8 representation. Therefore RequestParser is change to decode to QByteArray and anything that expects QString should make a second conversion from QByteArray to QString. Qt4 makes implicit conversion to QString using  fromAscii() and Qt5 uses fromUtf8(). That is why explicit QString::fromUtf8() is used for 'category'.